akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingGetStatsSpec.scala

```diff
@@ -109,17 +109,18 @@ abstract class ClusterShardingGetStatsSpec extends MultiNodeSpec(ClusterSharding
       join(second)
       join(third)
 
-      // make sure all nodes has joined
-      awaitAssert {
-        Cluster(system).sendCurrentClusterState(testActor)
-        expectMsgType[CurrentClusterState].members.size === 3
+      // make sure all nodes are up
+      within(10.seconds) {
+        awaitAssert {
+          Cluster(system).state.members.count(_.status == MemberStatus.Up) === 4
```

Comment:
Yes, before it got past here with nodes still Joining and then timed out later because getting all nodes up before sharding got started and in contact took time.